contrib/endpoints/src/api_manager/api_manager_impl.cc

```diff
@@ -76,16 +88,13 @@ utils::Status ApiManagerImpl::Init() {
     global_context_->cloud_trace_aggregator()->Init();
   }
 
-  if (!service_config_.empty()) {
-    std::string config_id;
-    if (AddConfig(service_config_, &config_id)) {
-      DeployConfigs({{config_id, 100}});
-
-      config_loading_status_ = utils::Status::OK;
-    } else {
-      config_loading_status_ =
-          utils::Status(Code::ABORTED, "Invalid service config");
+  if (!service_context_map_.empty()) {
+    for (auto it : service_context_map_) {
+      if (it.second->service_control()) {
+        it.second->service_control()->Init();
+      }
     }
+
     return config_loading_status_;
```

Comment:
If the constructor got the valid service_config text, then ApiManager doesn't need to create an instance of the ConfigManager.